spec/domain/sftp_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas.
+
+require 'spec_helper'
+
+describe Sftp do
+  let(:config) do
+    Config::Options.new(host: 'sftp.local',
+                        user: 'hitobito',
+                        password: 'password',
+                        private_key: 'private key',
+                        port: 22)
+  end
+
+  subject { Sftp.new(config) }
+
+  context 'with password' do
+    before { config.delete_field!(:private_key) }
+
+    it 'creates connection with password credential' do
+      session = double
```

Comment:
die session würde ich noch in ein let nehmen und mehr Sicherheit hätten wir mit einem `instance_double`